src/org/jetbrains/plugins/scala/lang/psi/types/api/TypeParameterType.scala

```diff
@@ -101,13 +101,8 @@ case class TypeParameterType(arguments: Seq[TypeParameterType],
 
   override val name = nameAndId._1
 
-  private var hash: Int = -1
-
-  override def hashCode: Int = {
-    if (hash == -1) {
-      hash = Seq(name, arguments, lowerType, upperType, psiTypeParameter).map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
```

Comment:
@niktrop should we expect a performance impact from seq/map/fold object allocations here or is it negligible because it's a one-time cost?